app/src/test/java/com/sebastianvm/mango/CoroutineTestExtension.kt

```diff
@@ -0,0 +1,30 @@
+package com.sebastianvm.mango
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.jupiter.api.extension.AfterAllCallback
+import org.junit.jupiter.api.extension.BeforeAllCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+
+class CoroutineTestExtension : BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
+    @OptIn(ExperimentalCoroutinesApi::class)
+    val dispatcher = UnconfinedTestDispatcher(name = "Default tests dispatcher")
+    lateinit var testScope: TestScope
+
+    override fun beforeEach(context: ExtensionContext?) {
+        testScope = TestScope(dispatcher)
```

Comment:
We cannot run `testScope.runTest` multiple times per test, so we need to create a new instance before each test to make sure parameterized tests work